src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabProjectSCMEvent.java

```diff
@@ -54,13 +57,61 @@ public String getSourceName() {
 
     @Override
     public boolean isMatch(@NonNull SCMNavigator navigator) {
-        return navigator instanceof GitLabSCMNavigator && ((GitLabSCMNavigator) navigator).getNavigatorProjects().contains(getPayload().getPathWithNamespace());
+        return navigator instanceof GitLabSCMNavigator && isMatch((GitLabSCMNavigator) navigator);
+    }
+
+    private boolean isMatch(@Nonnull GitLabSCMNavigator navigator) {
+        switch (getType()) {
+            case CREATED:
+                String projectPathWithNamespace = getPayload().getPathWithNamespace();
+                String projectOwner = getProjectOwnerFromNamespace(projectPathWithNamespace);
+                GitLabSCMNavigatorContext navigatorContext = new GitLabSCMNavigatorContext().withTraits(navigator.getTraits());
+                if(navigator.isGroup()) {
+                    // checks when project owner is a Group
+                    if(navigatorContext.wantSubgroupProjects()) {
+                        // can be a subgroup so needs to at least start with the project owner when subgroup projects are required
+                        return projectOwner.startsWith(navigator.getProjectOwner());
+                    } else {
+                        // when subgroup projects are not required, project owner should match project owner
+                        return projectOwner.equals(navigator.getProjectOwner());
+                    }
+                } else {
+                    // checks when project owner is a User
+                    if(navigatorContext.wantSubgroupProjects()) {
+                        // check if owner name is same as owner name stored when subgroup projects are required
+                        // this is done as username is not supplied in the system hook
+                        return navigator.getOwnerName().equals(getPayload().getOwnerName());
+                    } else {
+                        // check if username matches when subgroup projects are not required
+                        // project owner is derived from project namespace
+                        return projectOwner.equals(navigator.getProjectOwner());
+                    }
+                }
+            case UPDATED:
+                if(navigator.getNavigatorProjects().contains(getPayload().getPathWithNamespace())) {
+                    // TODO: Not sure if this is the way to do it, need to check
+                    return true;
+                }
+                break;
+            case REMOVED:
+                if(navigator.getNavigatorProjects().contains(getPayload().getPathWithNamespace())) {
+                    return true;
+                }
+                break;
+            default:
+                return false;
+        }
+        return false;
+    }
+
+    private String getProjectOwnerFromNamespace(String projectPathWithNamespace) {
```

Comment:
perhaps this method should be moved and made static as you also have the code here
https://github.com/jenkinsci/gitlab-branch-source-plugin/blob/0ca28a48aea1dbf9d87807c4d8de2912e1456819/src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMNavigator.java#L222